File: crypto/src/main/java/io/oversec/one/crypto/AbstractEncryptionParams.java
package io.oversec.one.crypto;

import android.content.Context;
import io.oversec.one.crypto.encoding.IXCoder;
import io.oversec.one.crypto.encoding.pad.AbstractPadder;


public abstract class AbstractEncryptionParams {

    private String mCoderId;
    private String mPadderId;

    private EncryptionMethod mEncryptionMethod;

    protected AbstractEncryptionParams(EncryptionMethod method, String coderId, String padderId) {
        mEncryptionMethod = method;
        mCoderId = coderId;
        mPadderId = padderId;
    }

    public EncryptionMethod getEncryptionMethod() {
        return mEncryptionMethod;
    }


    public String getCoderId() {
        return mCoderId;
    }

    public String getPadderId() {
        return mPadderId;
    }

    public void setXcoderAndPadder(IXCoder xcoder, AbstractPadder padder) {
        mCoderId = xcoder.getId();
        mPadderId = padder == null ? null : padder.getId();
    }

    public void setXcoderAndPadderIds(String xcoderId, String padderId) {
        mCoderId =  xcoderId;
        mPadderId = padderId;
    }


    public abstract boolean isStillValid(Context ctx);
}

File: crypto/src/main/java/io/oversec/one/crypto/AbstractEncryptionParams.kt
package io.oversec.one.crypto

import android.content.Context
import io.oversec.one.crypto.encoding.IXCoder
import io.oversec.one.crypto.encoding.pad.AbstractPadder


abstract class AbstractEncryptionParams protected constructor(
    val encryptionMethod: EncryptionMethod,
    coderId: String,
    padderId: String?
) {

    var coderId: String = coderId
        private set
    var padderId: String? = padderId
        private set


    fun setXcoderAndPadder(xcoder: IXCoder, padder: AbstractPadder?) {
        coderId = xcoder.id
        padderId = padder?.id
    }

    fun setXcoderAndPadderIds(xcoderId: String, padderId: String) {
        coderId = xcoderId
        this.padderId = padderId
    }

    abstract fun isStillValid(ctx: Context): Boolean
}
